src/MakingSense.AspNetCore.Authentication.SimpleToken/SimpleTokenAuthenticationHandler.cs

```diff
@@ -75,33 +84,39 @@ public static string ExtractToken(HttpRequest request)
 		/// <returns></returns>
 		protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
 		{
-			var token = ExtractToken(Request);
-
-			// If no token found, no further work possible
-			if (string.IsNullOrEmpty(token))
+			try
 			{
-				return AuthenticateResult.Skip();
-			}
+				var token = ExtractToken(Request);
 
-			var validationParameters = Options.TokenValidationParameters.Clone();
+				// If no token found, no further work possible
+				if (string.IsNullOrEmpty(token))
+				{
+					return AuthenticateResult.NoResult();
+				}
 
-			SecurityToken validatedToken;
-			var validators = Options.SecurityTokenValidatorsFactory();
-			foreach (var validator in validators)
-			{
-				if (validator.CanReadToken(token))
+				var validationParameters = Options.TokenValidationParameters.Clone();
+
+				var validators = Options.SecurityTokenValidatorsFactory();
+				foreach (var validator in validators)
 				{
-					var principal = validator.ValidateToken(token, validationParameters, out validatedToken);
-					var ticket = new AuthenticationTicket(principal, new AuthenticationProperties(), Options.AuthenticationScheme);
-					return AuthenticateResult.Success(ticket);
+					if (validator.CanReadToken(token))
+					{
+						var principal = validator.ValidateToken(token, validationParameters, out SecurityToken validatedToken);
+						var ticket = new AuthenticationTicket(principal, Scheme.Name);
+						return AuthenticateResult.Success(ticket);
+					}
 				}
-			}
 
-			// Ugly patch to make this method should to be async in order to allow result caching by caller
-			await DoneTask;
+				// Ugly patch to make this method should to be async in order to allow result caching by caller
+				await DoneTask;
 
-			// Not so nice, but AuthenticateResult.Fail does not allow us to show the error
-			throw new AuthenticationException("Authorization token has been detected but it cannot be read.");
+				// Not so nice, but AuthenticateResult.Fail does not allow us to show the error
+				throw new AuthenticationException("Authorization token has been detected but it cannot be read.");
```

Comment:
Does it work fine? should we not use `return AuthenticateResult.Fail("Authorization token has been detected but it cannot be read.")`?

Reply:
Yes, you are right, now could directly return the AuthenticationResult.Fail